packages/extension-api/src/extension-api.d.ts

```diff
@@ -4395,6 +4395,23 @@ declare module '@podman-desktop/api' {
      * @returns CliTool instance
      */
     export function createCliTool(options: CliToolOptions): CliTool;
+
+    /**
+     * given an id, return the corresponding CLI Tool
+     * @param id cli tool
+     */
+    export function getCliTool(id: string): CliTool | undefined;
+
+    /**
+     * All extensions currently known to the system.
+     */
+    export const all: readonly CliTool[];
```

Comment:
here I'm not sure. Because an extension B should not be able to receive the object provided by the extension A for security/responsability

you should receive some information (but a subset in readonly)

Reply:
updated, not sure what kind of subset you meant. I just returned all clitool properties excluding its methods